server/tests/services/user.test.js

```diff
@@ -206,4 +208,131 @@ describe('# User Service', () => {
         });
     });
   });
+
+  describe('# addDevice() | getDevice()', () => {
+    const baseUser = {
+      name: 'Jarryd Lee',
+      email: 'Jarryd@lee.com'
+    };
+
+    const device = new Device({
+      name: 'email',
+      type: 'email',
+      contactInformation: '1234567890'
+    });
+
+    let user;
+
+    before((done) => {
+      userService.createUser(baseUser)
+        .then((createdUser) => {
+          user = createdUser;
+          done();
+        });
+    });
+
+    it('should add and retrieve a users device', (done) => {
+      user.addDevice(device)
+        .then(savedUser => savedUser.getDevice(device.id))
+        .then((retrievedDevice) => {
+          expect(retrievedDevice.name).to.equal(device.name);
+          expect(retrievedDevice.type).to.equal(device.type);
+          expect(retrievedDevice.contactInformation).to.equal(device.contactInformation);
+          done();
+        });
+    });
+
+    it('should fail to retrieve a users device with incorrect deviceId', (done) => {
+      const fakeId = '1234567890abcdef123456789';
+      user.getDevice(fakeId)
+        .catch((err) => {
+          expect(err).to.exist;
+          expect(err.message).to.equal(`Device with ID ${fakeId} doesn\'t exist!`);
+          done();
+        });
+    });
+  });
+
+
+  describe('# removeDevice()', () => {
+    const baseUser = {
+      name: 'Jarryd Lee',
+      email: 'Jarryd@lee.com'
+    };
+
+    const device = new Device({
+      name: 'email',
+      type: 'email',
+      contactInformation: '1234567890'
+    });
+
+    let user;
+
+    before((done) => {
+      userService.createUser(baseUser)
+        .then(createdUser => createdUser.addDevice(device))
+        .then((updatedUser) => {
+          user = updatedUser;
+          done();
+        });
+    });
+
+    it('should remove a device from the user', (done) => {
+      userService.removeDevice(user, device.id)
+        .then((updatedUser) => {
+          expect(updatedUser).to.exist;
+          expect(updatedUser.devices).to.be.empty;
+          done();
+        });
+    });
+  });
+
+  describe('# sortDevices()', () => {
+    const baseUser = {
+      name: 'Jarryd Lee',
+      email: 'Jarryd@lee.com'
+    };
+
+    const baseDevice = {
+      name: 'email',
+      type: 'email',
+      contactInformation: '1234567890'
+    };
+
+    const device1 = new Device(baseDevice);
```

Comment:
I need the `id`s defined when I'm adding the device to the user in the `before` block as well as later to test equality. If I do the mapping, they are not created until after the before block as run, so I wouldn't have access to them.